pydantic/fields.py

```diff
@@ -70,6 +70,7 @@ class Field:
         'shape',
         'class_validators',
         'parse_json',
+        'self_validators',
```

Comment:
Do we really need this? Can't we use `whole_post_validators`?

Reply:
if we do need it, I think the name is confusing. Maybe `whole_validator`? 

Also, if we do need it, please set it to `None` in `__init__` so you don't need to use `hasattr()` later.

Reply:
Refactored this to minimal changes using whole pre validator